app/com/arpnetworking/kairos/client/models/KairosMetricNamesQueryResponse.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2017 Smartsheet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.kairos.client.models;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import net.sf.oval.constraint.NotNull;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Model class that represents a metric names response.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public final class KairosMetricNamesQueryResponse {
+    public List<String> getResults() {
+        return _results;
+    }
+
+    @JsonAnyGetter
+    public Map<String, Object> getOther() {
+        return _other;
+    }
+
+    private KairosMetricNamesQueryResponse(final Builder builder) {
+        _results = builder._results;
+        _other = builder._other;
+    }
+
+    private final List<String> _results;
+    private final Map<String, Object> _other;
```

Comment:
We should use `ImmutableList` and `ImmutableMap`.

Reply:
Done